bitcoin/src/network/mod.rs

```diff
@@ -175,13 +224,25 @@ impl Network {
 
     /// Returns the associated network parameters.
     pub const fn params(self) -> &'static Params {
-        const PARAMS: [Params; 4] = [
-            Params::new(Network::Bitcoin),
-            Params::new(Network::Testnet),
-            Params::new(Network::Signet),
-            Params::new(Network::Regtest),
-        ];
-        &PARAMS[self as usize]
+        match self {
+            Network::Bitcoin => &Params::BITCOIN,
+            Network::Testnet(TestnetVersion::V3) => &Params::TESTNET,
+            Network::Testnet(TestnetVersion::V4) => &Params::TESTNET4,
+            Network::Signet => &Params::SIGNET,
+            Network::Regtest => &Params::REGTEST,
```

Comment:
This logic (`Network` to `Params`) is duplicated a few times in the code: this one, `Params::new` and `Params::as_ref`. Not something this pr introduced, but an easy fix. `Params::new` (or any `const fn(Network) -> Params` in general) should be able to cover all needs from the other functions like returning a reference.